src/kudu/consensus/binlog.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <memory>
+
+#include "kudu/consensus/log_interface.h"
+
+namespace kudu {
+namespace consensus {
+
+class LogCache;
+
+class Binlog : public LogApi {
```

Comment:
Maybe keep Binlog implementation outside of Kudu. LogApi could be a virtual class that gives a set of API'a that any log handling class need to implement. During kuduconsensus initialization, we could pass an object of Binlog (as LogApi) which kuduconsensus can use to maintain the log.

Ignore if this is not how this is being designed